internal/impl/redpanda/functions.go

```diff
@@ -0,0 +1,156 @@
+// Copyright 2024 Redpanda Data, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package redpanda
+
+import (
+	"context"
+
+	"github.com/tetratelabs/wazero/api"
+)
+
+const (
+	noActiveTransform = int32(-1)
+	invalidBuffer     = int32(-2)
+)
+
+var transformHostFunctions = map[string]func(r *dataTransformEngine) interface{}{}
```

Comment:
The codebase switched to using `any` instead of `interface{}` a while back. Not sure why the linter didn't catch this... I'll look into re-enabling it.

Reply:
Ah oops, this is copied from the old wasm module - will update!